framework/src/consumption/lib/quicksight/quicksight-subscription.ts

```diff
@@ -0,0 +1,229 @@
+      'sso:GetProfile',
+      'sso:AssociateProfile',
+      'sso:DisassociateProfile',
+      'sso:ListProfiles',
+      'sso:ListDirectoryAssociations',
+      'sso:DescribeRegisteredRegions',
+    ];
+
+    if (props.authenticationMethod != QuickSightAuthenticationMethod.IAM_IDENTITY_CENTER) {
+      this.policyActions = this.policyActions.concat(
+        [
+          'ds:AuthorizeApplication',
+          'ds:UnauthorizeApplication',
+          'ds:CheckAlias',
+          'ds:CreateAlias',
+          'ds:DescribeDirectories',
+          'ds:DescribeTrusts',
+          'ds:DeleteDirectory',
+          'ds:CreateIdentityPoolDirectory',
+        ],
+      );
+    }
+
+    this.executionRole = new Role(this, 'Role', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      inlinePolicies: {
+        QuickSightSubscription: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: this.policyActions,
+              resources: ['*'],
+            }),
+          ],
+        }),
+      },
+    });
+
+    const timeout = props.executionTimeout ?? Duration.minutes(5);
+
+    const provider = new DsfProvider(this, 'CrProvider', {
+      providerName: 'QuickSightSubscriptionProvider',
+      onEventHandlerDefinition: {
+        depsLockFilePath: __dirname+'/resources/QuickSightSubscription/package-lock.json',
+        entryFile: __dirname+'/resources/QuickSightSubscription/index.mjs',
+        handler: 'index.onEventHandler',
+        environment: {
+          AUTHENTICATION_METHOD: props.authenticationMethod,
+          AWS_ACCOUNT_ID: props.awsAccountId,
+          EDITION: props.edition,
+          IDENTITY_REGION: props.identityRegion,
+        },
+        iamRole: this.executionRole,
+        timeout,
+      },
+      isCompleteHandlerDefinition: {
+        iamRole: this.executionRole,
+        handler: 'index.isCompleteHandler',
+        depsLockFilePath: __dirname+'/resources/QuickSightSubscription/package-lock.json',
+        entryFile: __dirname+'/resources/QuickSightSubscription/index.mjs',
+        timeout,
+        environment: {
+          AUTHENTICATION_METHOD: props.authenticationMethod,
+          AWS_ACCOUNT_ID: props.awsAccountId,
+          EDITION: props.edition,
+          IDENTITY_REGION: props.identityRegion,
+        },
+      },
+      queryInterval: Duration.seconds(10),
+      removalPolicy: this.removalPolicy,
+    });
+
+    this.serviceToken = provider.serviceToken;
+    this.submitLogGroup = provider.onEventHandlerLogGroup;
+    this.statusLogGroup = provider.isCompleteHandlerLog!;
+    this.cleanUpLogGroup = provider.cleanUpLogGroup;
+    this.submitFunction = provider.onEventHandlerFunction;
+    this.statusFunction = provider.isCompleteHandlerFunction!;
+    this.cleanUpFunction = provider.cleanUpFunction;
+    this.cleanUpRole = provider.cleanUpRole;
+
+  }
+
+
+  public createQuickSightSubscription() {
```

Comment:
can you bundle the custom resource directly in the constructor?

Reply:
ack